src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1813,6 +1816,7 @@ def create_vmss(vmss_name, resource_group_name, image,
                          if app_gateway_type == 'new' else None)
 
     # public IP is used by either load balancer/application gateway
+    public_ip_address_id = None
```

Comment:
This is the main fix, as w/o it the template builder will throw later for undefined `public_ip_address_id`